src/main/java/com/google/devtools/build/lib/exec/StandaloneTestStrategy.java

```diff
@@ -473,7 +474,7 @@ private static Spawn createXmlGeneratingSpawn(
       envBuilder.put("TEST_TOTAL_SHARDS", "0");
     }
     return new SimpleSpawn(
-        action,
+        new XmlTestGeneration(action),
```

Comment:
It's not obvious that changing the owner of the spawn to an action that isn't registered with Bazel is safe. Making `SimpleSpawn` non-final or adding yet another constructor to it would resolve this concern.

Reply:
The tests demonstrate that this isn't safe.

Reply:
Looks good to me, but could you explain the rationale for this PR in the description?